godot-codegen/src/generator/central_files.rs

```diff
@@ -37,6 +37,7 @@ pub fn make_sys_central_code(api: &ExtensionApi) -> TokenStream {
             // This will need refactoring if VariantType is changed to a real enum.
             #[doc(hidden)]
             pub fn from_sys(enumerator: crate::GDExtensionVariantType) -> Self {
+                #[allow(clippy::unnecessary_cast)]
```

Comment:
I was getting a warning here. Not sure if it has to do with which target is being built for.

Reply:
i'd use `#[expect(..)]` instead here to verify that this warning actually gets emitted

Reply:
Would this cause issues for any targets that aren't running into this warning?

Reply:
possibly yeah i guess, im not sure why it shows up for you but not the CI though?

Reply:
That I'm not sure, I was wondering if maybe someone more familiar may know.

Reply:
We cannot use `#[expect(..)]` yet because our MSRV is 1.80, but the lint was [only introduced in 1.81](https://blog.rust-lang.org/2024/09/05/Rust-1.81.0.html#expectlint).

Anyway, whether this is needed should be a separate discussion so we do it consistently, not randomly start adding it in some PRs. Personally, I see it adding some benefit, but also extra maintenance to keep conditional compilation happy (already a pain with unused variables), while preventing very few bugs in practice. As such I'm not sure if it's something I want to start doing.

Reply:
Alright, in that case for now, I'll remove this line.

Reply:
Removed. Will open up something separate for this where that discussion can go.